doc/style-guide-testing.md

```diff
@@ -0,0 +1,467 @@
+        * Emit "no value"
+* When receiving "update" event (we intend to be broadcasting)
+    * If scope "even" is truthy
+        * Do not allow the event through
+    * Otherwise
+        * Do nothing
+* When receiving "dirty" event (we intend to be emitting)
+    * If scope "number" is a multiple of seven
+        * Do not allow the event through
+    * Otherwise
+        * Do nothing
+
+That's pretty convoluted and will require some thought when making tests.  Here's the test.  Can you see the things that I am missing?  You'll see a lot of code in the `beforeEach()` blocks everywhere.  This test is carefully structured to allow for future updates and additional tests to be trivial to implement.
+
+    /*global describe, inject*/
+    describe('myCustomDirective', function () {
+        var compile, $parent, rootScope;
+
+        beforeEach(module('app'));
+        beforeEach(inject(function ($compile, $rootScope) {
+            rootScope = $rootScope;
+            $parent = $rootScope.$new();
+            compile = function (html) {
+                $compile(angular.element(html))($parent);
+                return $parent.$$childHead;
+            };
+        }));
+        describe('initial compile', function () {
+            var allowNoValueEvent;
+
+            // By default we do not want the "no value" event emitted
+            beforeEach(function () {
+                allowNoValueEvent = false;
+                $parent.on('no value', function () {
+                    expect(allowNoValueEvent).toBe(true);
+                });
+            });
+
+            describe('when passed a value', function () {
+                var deferred, $scope, serviceSpy;
+
+                beforeEach(inject(function ($provide, $q) {
+                    $parent.theValue = {};
+                    deferred = $q.defer();
+                    serviceSpy = jasmine.createSpy('serviceSpy').andReturn(deferred.promise);
+                    $provide.value('aService', serviceSpy);
+                    $scope = compile('<div my-custom-directive="theValue"></div>');
+                }));
+                it('called the service and sent the necessary value', function () {
+                    expect(serviceSpy).toHaveBeenCalledWith($parent.theValue);
+                });
+                describe('when fulfilled with an even number', function () {
+                    beforeEach(function () {
+                        deferred.resolve(6);
+                        $rootScope.$apply();
+                    });
+                    it('sets only the "even" property', function () {
+                        expect($scope.even).toBe(true);
+                        expect($scope.odd).toBe(undefined);
+                        expect($scope.number).toBe(undefined);
+                        expect($scope.error).toBe(undefined);
+                    });
+                });
+                describe('when fulfilled with an odd number', function () {
```

Comment:
Style guide says to prefere double quotes.  I get some of the other describes and its since they include double quotes in the string.

Is it more important to be consistent throughout a file?